config/clients/js/template/common.mustache

```diff
@@ -164,18 +203,40 @@ export async function attemptHttpRequest<B, R>(
         throw new FgaApiAuthenticationError(err);
       } else if (status === 404) {
         throw new FgaApiNotFoundError(err);
-      } else if (status === 429 || status >= 500) {
-        if (iterationCount >= config.maxRetry) {
+      } else if (status === 429 || (status >= 500 && status !== 501)) {
```

Comment:
Can you also make sure we are retrying on network errors? This will have to be defined above